tck/src/main/java/ee/jakarta/tck/data/web/validation/PersistenceTests.java

```diff
@@ -0,0 +1,238 @@
+                new Rectangle("RECT-003", 1L, 1L, 1, 1) //positive
+                );
+        rectangles.saveAll(validRects);
+        
+       resultRects = getResults();
+        assertEquals(4, resultRects.size(), "Number of results was incorrect");
+        for(int i = 0; i < resultRects.size(); i++) {
+            assertEquals("RECT-00" + i, resultRects.get(i).getId(), "Result list did not contain the correct object");
+        } 
+    }
+    
+    @Assertion(id = "133", strategy="Ensure that entities that violate the constraints cannot be inserted into the database")
+    public void testSaveWithInvalidConstraints() {
+        ConstraintViolationException resultingException;
+        
+        // Save
+        Rectangle invalidRect = new Rectangle("RECT-010", -1L, -1L, -1, 0); //Min values
+        
+        resultingException = assertThrows(ConstraintViolationException.class, () -> {
+            rectangles.save(invalidRect);
+        });
+        assertEquals(0, rectangles.count(), "No rectangles should have presisted to database while violating constraints.");
+        assertEquals(4, resultingException.getConstraintViolations().size(), "Incorrect number of constraint violations.");
+        
+        // Save All
+        List<Rectangle> invalidRects = List.of(
+                new Rectangle("", 5L, 5L, 5, 5), //blank id
+                new Rectangle("RECT-012", 1801L, 5L, 5, 5), //Max x
+                new Rectangle("RECT-013", 5L, 1001L, 5, 5), //Max y
+                new Rectangle("RECT-014", 5L, 5L, 121, 5), //Max width
+                new Rectangle("RECT-014", 5L, 5L, 5, 81) //Max height
+                );
+        
+        resultingException = assertThrows(ConstraintViolationException.class, () -> {
+            rectangles.saveAll(invalidRects);
+        });
+        assertEquals(0, rectangles.count(), "No rectangles should have presisted to database while violating constraints.");
+        assertEquals(5, resultingException.getConstraintViolations().size(), "Incorrect number of constraint violations.");        
+    }
+    
+    @Assertion(id = "133", strategy="Ensure that entities that do not violate the constraints can be successfully updated in the database")
+    public void testUpdateWithValidConstraints() {
+        List<Rectangle> resultRects;
+        
+        // Save
+        Rectangle validRect = new Rectangle("RECT-020", 5L, 5L, 5, 5);
+        rectangles.save(validRect);
+        
+        assertEquals(1, rectangles.count(), "Number of results was incorrect");
+        
+        // Update
+        Rectangle updatedRect = new Rectangle("RECT-020", 10L, 10L, 10, 10);
+        rectangles.save(updatedRect);
+        
+        assertEquals(1, rectangles.count(), "Number of results was incorrect");
+        
+        // Verify
+        resultRects = getResults();
+        Rectangle resultRect = resultRects.get(0);
+        assertRectangleFields(updatedRect, resultRect);      
```

Comment:
Although in practice this probably works in almost all cases, Jakarta makes no guarantees about the state of an entity parameter that is supplied to a save operation after the method, so we shouldn't be asserting based on it. The result of save needs to be used instead of the parameter supplied to it,
```suggestion
        updatedRect = rectangles.save(updatedRect);
        
        assertEquals(1, rectangles.count(), "Number of results was incorrect");
        
        // Verify
        resultRects = getResults();
        Rectangle resultRect = resultRects.get(0);
        assertRectangleFields(updatedRect, resultRect);      
```